### PR TITLE
Move to CircleCI

### PR DESCRIPTION
**Pull request by wtrocki**
### Motivation

https://issues.jboss.org/browse/AEROGEAR-1951

### Progress

- [x] Link core library
- [x] Migrate to new format for circle ci
- [x] Remove travis configuration
- [x] Add development script for supporting Visual Studio based development
- [x] Add circle cache for improving build speed 
- [x] Add style check (fail on style first)
- [x] Configure new version of ruby

## Notes

Travis is reported as hanging but it's actually disabled and it will not appear for the new PR's

### CODE REVIEW

**Comment by david-martin** [comment]
@wtrocki should travis be disabled on this repo now?

File: example/AeroGearSdkExampleTests/AeroGearSdkExampleTests.swift [hunk 1]
```diff
@@ -3,31 +3,30 @@
...
         // Put setup code here. This method is called before the invocation of each test method in the class.
     }
-    
+
     override func tearDown() {
         // Put teardown code here. This method is called after the invocation of each test method in the class.
         super.tearDown()
     }
-    
+
```
Comment (david-martin) [inline example/AeroGearSdkExampleTests/AeroGearSdkExampleTests.swift]: @wtrocki are there Core SKD unit tests that could/should be added here?

**Comment by wtrocki** [comment]
I disabled travis, but this PR will still show build.

I used travis and circle CI to in this PR to compare build times. Circle were building immediately and finishing in 3 min. Travis builds were taking around 1.5 hour to start.

File: example/AeroGearSdkExampleTests/AeroGearSdkExampleTests.swift (hunk 1 above)
Comment (wtrocki) [reply example/AeroGearSdkExampleTests/AeroGearSdkExampleTests.swift]: I have separate PR for this. This PR is more about CI and local development.
This 2 tests are left to see if unit testing works.
Trying to keep things small to encourage more reviews :)

Comment (david-martin) [reply example/AeroGearSdkExampleTests/AeroGearSdkExampleTests.swift]: OK. Approving, but would like to get a review from someone else that has done Swift dev or used circle ci.

**Comment by wtrocki** [comment]
Closing as I would like to created another PR with the actual code and this is blocking me.

**Pull request merged**